lib/controllers/coinstac/consortia.js

```diff
@@ -17,31 +27,143 @@ internals.consortiaSchema = joi.object().keys({
 });
 
 internals.savedConsortiaSchema = internals.consortiaSchema.keys({
+    dbUrl: joi.string().required(),
     _id: joi.string().required(),
     _rev: joi.string().required()
 });
 
-internals.createConsortiaDb = (name, PouchW) => {
-    const defaults = consortiaDbOptions;
-    let pouchWConfig;
+/**
+ * Set the security on the consortiaDB to be more relaxed
+ * Only takes action when config.coinstac.pouchdb.cloudant is defined
+ * @param  {object} db the pouch-wrapper instance
+ * @return {Promise}    Resolves to the response from the cloudant API
+ */
+internals.setConsortiumDbSecurity = (db) => {
+    const security = {
+        nobody: [
+            '_reader',
+            '_writer'
+        ]
+    };
 
-    //TODO: return name of db for ease of location
-    name = defaults.name + '-' + _.kebabCase(name.toLowerCase());
-    if (consortiaDbOptions.path) {
-        pouchWConfig = {
-            path: defaults.path,
-            name: name
-        };
-    } else if (consortiaDbOptions.conn) {
-        pouchWConfig = {
-            name: name,
-            conn: defaults.conn
+    if (isCloudant) {
+        return cloudantClient.mergeSecurity(db.name, security);
+    } else {
+        return Bluebird.resolve();
+    }
+};
+
+/**
+ * compute average of values
+ * @param  {array} values array of objects with `value` and `weight` properties
+ * @return {object}       object with a `value` and `weight` property
+ */
+internals.average = (values) => {
+    return _.reduce(values, (res, value) => {
+        const weightedResVal = (res.value * res.weight);
+        const weightedVal = (value.value * value.weight);
+        const numerator = weightedResVal + weightedVal;
+        const denominator = res.weight + value.weight;
+        res.value = numerator / denominator;
+        res.weight = denominator;
+        return res;
+    });
+};
+
+/**
+ * Add listeners to newly created consortiumDb
+ * @param  {object} db     pouchDB or pouchWrapper insance
+ * @param  {server} server server instance
+ * @return {null}           nothing
+ */
+internals.watchConsortiumDb = (db, server) => {
```

Comment:
Eventually, I suspect that we will want to use a completely separate service to watch consortium DBs, but for now, I thought it would be good to keep logic contained within this function. As it grows in complexity, we can move it to a separate file.